google_cloudsql/postgres/main.tf

```diff
@@ -0,0 +1,138 @@
+/**
+ * # gcp-postgres
+ * Creates a PostgreSQL instance within GCP using Cloud SQL
+ */
+
+# https://www.terraform.io/docs/providers/google/r/sql_database_instance.html
+#
+# There is no need for failover replicas for regional PostgreSQL database
+# as that is provided by default.
+# It is possible to create one or more read replicas for regional PostgreSQL
+# database. For that see
+# https://github.com/terraform-providers/terraform-provider-google/issues/1773#issuecomment-408556558
+# for an example.
+
+locals {
+  default_database_name = "${var.application}-${var.realm}-${var.environment}-${var.instance_version}"
+  database_name         = coalesce(var.custom_database_name, local.default_database_name)
+  tier                  = coalesce(var.tier_override, "db-custom-${var.db_cpu}-${var.db_mem_gb * 1024}")
+}
+
+resource "google_sql_database_instance" "master" {
+  provider = google-beta
+  name     = local.database_name
+  project  = var.project_id
+  region   = var.region
+
+  database_version = var.database_version
+
+  settings {
+    tier              = local.tier
+    availability_type = var.availability_type
+    disk_size         = var.disk_size
```

Comment:
I believe `disk_autoresize` is optional but defaults true. In the documentation it says not to set disk_size since once auto resize kicks in terraform will try to resize the disk to disk_size value.

Reply:
will fix

Reply:
So does this mean we just default to disk size (10GB) and let cloudsql handle the auto resize?

Reply:
yeah, seems to be what terraform suggests